File: features/user/updateinactivephone/src/main/java/com/tokopedia/updateinactivephone/usecase/ValidateUserDataUseCase.java
package com.tokopedia.updateinactivephone.usecase;

import android.content.Context;

import com.tokopedia.abstraction.common.utils.GraphqlHelper;
import com.tokopedia.graphql.data.model.GraphqlRequest;
import com.tokopedia.graphql.domain.GraphqlUseCase;
import com.tokopedia.updateinactivephone.R;
import com.tokopedia.updateinactivephone.model.response.GqlValidateUserDataResponse;
import com.tokopedia.updateinactivephone.subscriber.ValidateUserDataSubscriber;

import java.util.HashMap;
import java.util.Map;

import static com.tokopedia.updateinactivephone.common.UpdateInactivePhoneConstants.OPERATION_NAME.VALIDATE_USER_DATA;
import static com.tokopedia.updateinactivephone.common.UpdateInactivePhoneConstants.QUERY_CONSTANTS.EMAIL;
import static com.tokopedia.updateinactivephone.common.UpdateInactivePhoneConstants.QUERY_CONSTANTS.PHONE;
import static com.tokopedia.updateinactivephone.common.UpdateInactivePhoneConstants.QUERY_CONSTANTS.USER_ID;

public class ValidateUserDataUseCase {

    private GraphqlUseCase graphqlUseCase;
    private Context context;

    public ValidateUserDataUseCase(Context context) {
        graphqlUseCase = new GraphqlUseCase();
        this.context = context;
    }

    public void unsubscribe() {
        if (graphqlUseCase != null) {
            graphqlUseCase.unsubscribe();
        }
    }

    public void execute(String phone, String email, String userId, ValidateUserDataSubscriber validateUserDataSubscriber) {
        graphqlUseCase.clearRequest();
        Map<String, Object> variables = new HashMap<>();

        variables.put(PHONE, phone);
        variables.put(EMAIL, email);
        variables.put(USER_ID, Integer.parseInt(userId));

        GraphqlRequest graphqlRequest = new GraphqlRequest(
                GraphqlHelper.loadRawString(context.getResources(), R.raw.query_validate_user_data),
                GqlValidateUserDataResponse.class,
                variables, VALIDATE_USER_DATA, false);

        graphqlUseCase.addRequest(graphqlRequest);
        graphqlUseCase.execute(validateUserDataSubscriber);
    }
}

File: features/user/updateinactivephone/src/main/java/com/tokopedia/updateinactivephone/usecase/ValidateUserDataUseCase.kt
package com.tokopedia.updateinactivephone.usecase

import android.content.Context

import com.tokopedia.abstraction.common.utils.GraphqlHelper
import com.tokopedia.graphql.data.model.GraphqlRequest
import com.tokopedia.graphql.domain.GraphqlUseCase
import com.tokopedia.updateinactivephone.R
import com.tokopedia.updateinactivephone.common.UpdateInactivePhoneConstants.OPERATION_NAME.Companion.VALIDATE_USER_DATA
import com.tokopedia.updateinactivephone.common.UpdateInactivePhoneConstants.QUERY_CONSTANTS.Companion.EMAIL
import com.tokopedia.updateinactivephone.common.UpdateInactivePhoneConstants.QUERY_CONSTANTS.Companion.PHONE
import com.tokopedia.updateinactivephone.common.UpdateInactivePhoneConstants.QUERY_CONSTANTS.Companion.USER_ID
import com.tokopedia.updateinactivephone.data.model.response.GqlValidateUserDataResponse
import com.tokopedia.updateinactivephone.viewmodel.subscriber.ValidateUserDataSubscriber

import java.util.HashMap


class ValidateUserDataUseCase(private val context: Context) {

    private val graphqlUseCase: GraphqlUseCase?

    init {
        graphqlUseCase = GraphqlUseCase()
    }

    fun unsubscribe() {
        graphqlUseCase?.unsubscribe()
    }

    fun execute(phone: String, email: String, userId: String, validateUserDataSubscriber: ValidateUserDataSubscriber) {
        graphqlUseCase!!.clearRequest()
        val variables = HashMap<String, Any>()

        variables[PHONE] = phone
        variables[EMAIL] = email
        variables[USER_ID] = Integer.parseInt(if (userId != "") userId else "0")

        val graphqlRequest = GraphqlRequest(
                GraphqlHelper.loadRawString(context.resources, R.raw.query_validate_user_data),
                GqlValidateUserDataResponse::class.java,
                variables, VALIDATE_USER_DATA, false)

        graphqlUseCase.addRequest(graphqlRequest)
        graphqlUseCase.execute(validateUserDataSubscriber)
    }
}
